include/aspect/particle/integrator/euler.h

```diff
@@ -77,6 +86,14 @@ namespace aspect
             * The forward euler integrator does not need any intermediate storage space.
             */
           static const unsigned int n_integrator_properties = 0;
+
+        private:
+          /**
+           * Cache whether the geometry model has periodic boundaries. If so,
+           * we need to adjust the particle position if it crosses a periodic
+           * boundary.
+           */
+          bool has_periodic_boundary;
```

Comment:
I think it would be useful to rename the variable as geometry_has_periodic_boundary because it's a property of the geometry, not of the Euler/RK2/RK4 integrator.

I also thought some about whether it's really useful to store this sort of information in the class at all. It's really not a property of the integrator. You're just caching this information because you think it's expensive to re-compute over and over. But is this actually true? My preference would be to just compute it in the one place (in each class) where you actually use it.